configuration-metadata/spring-boot-configuration-processor/src/test/java/org/springframework/boot/configurationprocessor/test/ItemMetadataAssert.java

```diff
@@ -28,34 +31,33 @@
  * AssertJ assert for {@link ItemMetadata}.
  *
  * @author Stephane Nicoll
+ * @author Stefano Cordio
  */
 public class ItemMetadataAssert extends AbstractAssert<ItemMetadataAssert, ItemMetadata>
 		implements AssertProvider<ItemMetadataAssert> {
 
-	private static final Objects objects = Objects.instance();
-
 	public ItemMetadataAssert(ItemMetadata itemMetadata) {
 		super(itemMetadata, ItemMetadataAssert.class);
-		objects.assertNotNull(this.info, itemMetadata);
+		isNotNull();
 	}
 
 	public ItemMetadataAssert isProperty() {
-		objects.assertEqual(this.info, this.actual.isOfItemType(ItemType.PROPERTY), true);
+		extracting((actual) -> actual.isOfItemType(ItemType.PROPERTY)).isEqualTo(true);
```

Comment:
I favored the usage of `extracting` here and in the other methods to leverage the [internal propagation](https://github.com/assertj/assertj/blob/c97dde272816c17716326d791f2a6ba8e31453ef/assertj-core/src/main/java/org/assertj/core/api/AbstractAssert.java#L1234) of the assertion state.

In case that isn't a required feature, such lines could also be written as:

```suggestion
		Assertions.assertThat(this.actual.isOfItemType(ItemType.PROPERTY)).isTrue();
```

and the private `extracting` method could be deleted.